R/format_PET.R

```diff
@@ -0,0 +1,1200 @@
+                  NestboxID = toupper(.data$NoNestBox),
+                  LocationID = .data$NestboxID,
+                  BroodID = as.character(paste(.data$BreedingSeason, .data$TheLineOfArtificialNestBoxes,
+                                               .data$NestboxID, .data$StartDateOfLaying1May1,
+                                         sep = "_")),
+                  Species = species_codes[which(species_codes$SpeciesID == 14640), ]$Species,
+                  FemalesRingSeries = stringr::str_replace(.data$FemalesRingSeries,
+                                                           pattern = " ", replacement = ""),
+                  FemaleID = if_else(!is.na(.data$FemalesRingSeries) & !is.na(.data$FemalesRing),
+                                     as.character(paste0(.data$FemalesRingSeries, .data$FemalesRing)),
+                                     NA_character_),
+                  MalesRingSeries = stringr::str_replace(.data$MalesRingSeries,
+                                                           pattern = " ", replacement = ""),
+                  MaleID = if_else(!is.na(.data$MalesRingSeries) & !is.na(.data$MalesRing),
+                                     as.character(paste0(.data$MalesRingSeries, .data$MalesRing)),
+                                     NA_character_),
+                  SeriesOfNestlingsRings = stringr::str_replace_all(.data$SeriesOfNestlingsRings,
+                                                                    pattern = " ", replacement = ""),
+                  NestlingRings = stringr::str_replace_all(.data$NestlingRings,
+                                                           pattern = " ", replacement = ""),
+                  StartDateOfLaying1May1 = as.numeric(.data$StartDateOfLaying1May1)) %>%
+    #### Remove columns which we do not store in the standardized format
+    dplyr::select(-.data$FemalesRingSeries,
+                  -.data$FemalesRing,
+                  -.data$MalesRingSeries,
+                  -.data$MalesRing,
+                  -.data$NoString,
+                  -.data$NoNestBox,
+                  -.data$NestedEarlier18,
+                  -.data$NestedEarlier23) %>%
+    #### Reorder columns
+    dplyr::select(.data$BreedingSeason,
+                  .data$Species,
+                  .data$PopID,
+                  everything()) %>%
+    dplyr::distinct()
+
+
+  message("Importing primary data Pied flycatcher ...")
+
+  fichyp_data <- readxl::read_excel(path =  paste0(db, "/PET_PrimaryData.xls"),
+                                    na = c("NA", "nA"),
+                                    sheet = 1) %>% ## Note: This does not work on Mac: sheet = "Pied flycatcher"
+    janitor::clean_names(case = "upper_camel") %>%
+    janitor::remove_empty(which = "rows") %>%
+    #### Convert to corresponding format and rename
+    dplyr::mutate(PopID = "PET",
+                  BreedingSeason = as.integer(.data$Year),
+                  NestboxID = toupper(.data$NoNestBox),
```

Comment:
Again as above, there are some Cyrillic characters there that may cause problems (they're not obvious because they look the same as latin!). Below code should fix it:
```suggestion
                  NestboxID = stringi::stri_trans_general(.data$NoNestBox, "cyrillic-latin"),
```